hawkular-wildfly-agent/src/main/java/org/hawkular/agent/monitor/extension/OperationSubsystemStart.java

```diff
@@ -37,22 +37,34 @@ public void execute(OperationContext opContext, ModelNode model) throws Operatio
             MonitorService service = (MonitorService) serviceRegistry.getRequiredService(name).getValue();
 
             boolean restart = model.get("restart").asBoolean(false);
-            if (restart) {
-                LOGGER.debug("Asked to restart the Hawkular Monitor service. Will stop it, then restart it now.");
-                service.stopMonitorService();
-            } else {
-                LOGGER.debug("Asked to start the Hawkular Monitor service");
-            }
-            if (!service.isMonitorServiceStarted()) {
-                Thread newThread = new Thread(new Runnable() {
-                    @Override
-                    public void run() {
+            long delay = model.get("delay").asLong(0L);
+
+            Thread newThread = new Thread(new Runnable() {
+                @Override
+                public void run() {
+                    if (delay > 0) {
+                        try {
+                            Thread.sleep(delay);
+                        } catch (Exception e) {
+                            // just continue with start

```

Comment:
This needs to abort and not continue. What is probably happening here is the sleep got an InterruptedException, which usually means the app server has been sent a shutdown signal and all threads are being told to shutdown. So I would re-thrown the exception here which helps the app server shutdown.